clients/java/src/test/java/io/zeebe/client/CredentialsTest.java

```diff
@@ -0,0 +1,90 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client;
+
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.grpc.ServerInterceptors;
+import io.grpc.testing.GrpcServerRule;
+import io.zeebe.client.impl.BearerTokenCredentialsProvider;
+import io.zeebe.client.impl.ZeebeClientBuilderImpl;
+import io.zeebe.client.impl.ZeebeClientImpl;
+import io.zeebe.client.util.RecordingGatewayService;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.Consumer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class CredentialsTest {
+
+  @Rule public final GrpcServerRule serverRule = new GrpcServerRule();
+  @Parameter public Consumer<ZeebeClientBuilder> clientConfigurator;
+
+  private ZeebeClient client;
+  private ZeebeClientBuilderImpl builder;
+
+  @Parameters
+  public static Collection<Consumer<ZeebeClientBuilder>> parameters() {
+    return Arrays.asList(
+        builder ->
+            builder
+                .usePlaintext()
+                .credentialsProvider(
+                    new BearerTokenCredentialsProvider(
+                        CredentialsTest.class.getClassLoader().getResource("creds").getPath())),
+        builder ->
+            builder
+                .usePlaintext()
+                .credentialsProvider(
+                    headers -> {
+                      final Key<String> key =
+                          Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
+                      headers.put(key, "Bearer differentToken");
+                    }));
+  }
+
+  @Before
+  public void setUp() {
+    final RecordingGatewayService gatewayService = new RecordingGatewayService();
+    serverRule
+        .getServiceRegistry()
+        .addService(ServerInterceptors.intercept(gatewayService, new JwtInterceptor()));
+    builder = new ZeebeClientBuilderImpl();
+  }
+
+  @After
+  public void tearDown() {
+    if (client != null) {
+      client.close();
+      client = null;
+    }
+  }
+
+  @Test
+  public void shouldUseCallCredentials() {
+    clientConfigurator.accept(builder);
+    client = new ZeebeClientImpl(builder, serverRule.getChannel());
+    client.newTopologyRequest().send().join();
```

Comment:
Can we do simpler? Assume I'm stupid and never want to think :sweat_smile: So I'd like not to know how `JwtInterceptor` works...can we have a recording interceptor which just records the token? And here I just assert it got the right token that I passed to it?

Actually, come to think of it, we want to test that the headers we modified are applied to the request, and nothing else. Don't even care about the token.